### PR TITLE
Variable nuke disk generator locations

### PR DESCRIPTION
**Pull request by Lumipharon**
## About The Pull Request
Random nuke disk locations.

Maps can now have more than three possible nuke disk gen locations. The game will randomly choose three locations to actually spawn the disk gens.

This occurs round start for Nuke war, or when the ship crashes during Crash (thus avoiding disk gens getting crushed by the Cant).

For this initial PR I've just added one or two new spots for the larger maps (and RO). I'll leave it to Grayson or someone else to take a proper look in the future, as I'm sure more spots can be added, but the existing spots may need to be moved around as part of that.
## Why It's Good For The Game
Players are very static with how the play the game. They find optimal paths and there's little reason to deviate from them. This means gameplay tends to happen in the exact same areas every single game.

This change should help spread gameplay out across the map more, and add some variety with different sets of disk locations per game.
## Changelog
:cl:
add: Nuke disk generators are now randomised in location, based on a list of possible spawn points
/:cl:

### CODE REVIEW

**Comment by Mrrpip** [comment]
Oooh, fancy.

That said, how does it work exactly?

Can it work with nightmare inserts for example? Or is it restricted to 'guaranteed' map areas?

**Comment by Lumipharon** [comment]
> Oooh, fancy.
> 
> That said, how does it work exactly?
> 
> Can it work with nightmare inserts for example? Or is it restricted to 'guaranteed' map areas?

It works with modulars, you can see the map diff where I'm changing them

**Review by RipGrayson — APPROVED** [review]
mapping is fine

**Comment by Arturlang** [comment]
Honestly at this point it'd be better to just have generic computers around that CAN either have intel or nuke disk programs

**Comment by Lumipharon** [comment]
> Honestly at this point it'd be better to just have generic computers around that CAN either have intel or nuke disk programs

It was considered, but intel computers can be close to each other without problem, where as its very undesireable for disk computers to be too close to each other.

File: code/game/objects/machinery/computer/nuke_disk_generator.dm [hunk 1]
```diff
@@ -200,3 +200,20 @@
 	icon_state = "nuke_blue"
 	disk_type = /obj/item/disk/nuclear/blue
 	disk_color = "blue"
+
+//list of disk gens we want to spawn for nuke related gamemodes. Could just use subtypes, but using a list in case more gen related stuff is added in the future
+GLOBAL_LIST_INIT(nuke_disk_generator_types, list(/obj/machinery/computer/nuke_disk_generator/red, /obj/machinery/computer/nuke_disk_generator/green, /obj/machinery/computer/nuke_disk_generator/blue))
+
+/obj/structure/computer3frame/nuke_disk_candidate
```
Comment (TiviPlus) [inline code/game/objects/machinery/computer/nuke_disk_generator.dm:207]: why is this a computer3frame child

**Review by TiviPlus — CHANGES_REQUESTED** [review]
rest fine

File: code/game/objects/machinery/computer/nuke_disk_generator.dm (hunk 1 above)
Comment (Lumipharon) [reply code/game/objects/machinery/computer/nuke_disk_generator.dm]: Copy pasted for stats/icon, forgot to change it. Updated now.

**Pull request merged**